src/ui/public/visualize/visualize.js

```diff
@@ -50,9 +52,19 @@ uiModules
       const responseHandler = getHandler(responseHandlers, $scope.vis.type.responseHandler);
 
       $scope.fetch = _.debounce(function () {
+
         // searchSource is only there for courier request handler
         requestHandler($scope.vis, $scope.appState, $scope.uiState, $scope.savedObj.searchSource)
-          .then(resp => responseHandler($scope.vis, resp), e => {
+          .then(requestHandlerResponse => {
+            if ($scope.requestHandlerResponse && $scope.requestHandlerResponse === requestHandlerResponse) {
+              //we do not need to call the response handler, since there have been no data changes
```

Comment:
In practice, this saves a new tabify on each ui-state change which happens all the time when panning/zooming the map. this also saves processing of the geojson on each change.

by skipping this, we can also rely on the status.data===false flag to NOT recreate the leaflet layer, because we know the data hasn't changed. This is an important part to keep the map responsive. If we do rerun the reponseHandler, the data flag becomes `true` (even though conceptually).

This doesn't really affect other visualizations because they don't push on the ui-state frequently.